core/controllers/classifier.py

```diff
@@ -79,7 +79,10 @@ def verify_signature(message, vm_id, received_signature):
     if secret is None:
         return False
 
-    generated_signature = generate_signature(secret, message)
+    if message is None:
```

Comment:
If that's the case then it also modifies content sent from `Oppia-ml`. If this is your final decision then I will modify the wiki page and `Oppia-ml`'s code on this part.